### DISCUSSION TITLE
Record type definition by developer name in `data import tree` command

### DISCUSSION THREAD

**Discussion by RemiLeGuin** [forcedotcom/cli#2753] [Ideas]
When inserting data in an org with the `data import tree` [command (beta or legacy)](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_reference.meta/sfdx_cli_reference/cli_reference_data_commands_unified.htm#cli_reference_data_import_beta_tree_unified), it is possible to set a specific record type ID for the records if there are several. However, setting the record type developer name would be much easier as we cannot know the record type ID unless we request it first and modify the data to insert in the JSON file.

This could have the following format:
```
{
    "records": [
        {
            "attributes": {
                "type": "Account",
                "referenceId": "AccountRef1"
            },
            "RecordType.DeveloperName" : "B2Baccount",
            "Name": "Guimini",
            "Type": "Consulting",
            "Website": "https://www.guimini.com"
        }
    ]
}
```

Or this one:
```
{
    "records": [
        {
            "attributes": {
                "type": "Account",
                "referenceId": "AccountRef1"
            },
            "RecordType" : {
                "DeveloperName": "B2Baccount"
            },
            "Name": "Guimini",
            "Type": "Consulting",
            "Website": "https://www.guimini.com"
        }
    ]
}
```

Thanks @mshanemc for the suggestion to create an idea for it!

**Comment by mshanemc** [comment]
Yeah, I *think* we'd need to have the export command translate between RecordType Ids and developerNames.  We could pretty easily detect if your query included recordType and do **one more query** to get the names/ID mapping.

Then, we could do the opposite on load (quick scan of data for RecordType, and replace them with IDs from the new org like we do refs).
